File: InsertionSort/insertionSort.js

const insertionSort = (A) => {
    const n = A.length;
    let temp;
    for (let i = 0; i < n; i++) {
        for (let j = 1; j < n; j++) {
            if (A[j] < A[j - 1]) {
                temp = A[j - 1]
                A[j - 1] = A[j]
                A[j] = temp
            }
        }
    }
    return A;
}

A = []
for (let i = 0; i < 10; i++) {
    A.push(Math.floor(Math.random() * 30) + 1)
}
console.log(A)
console.log(insertionSort(A))

File: InsertionSort/insertion_sort.py
import numpy as np 

A = list(np.random.randint(0, 30, (1, 10)).reshape(-1))
print(A)

def insertion_sort(A):
    n = len(A)
    for i in range(n):
        for val in range(1,n):
            if A[val] < A[val-1]:
                A[val-1], A[val] = A[val], A[val-1]

    return A

print(insertion_sort(A))